probe/appclient/report_publisher.go

```diff
@@ -8,18 +8,37 @@ import (
 // A ReportPublisher uses a buffer pool to serialise reports, which it
 // then passes to a publisher
 type ReportPublisher struct {
-	publisher Publisher
+	publisher  Publisher
+	noControls bool
 }
 
 // NewReportPublisher creates a new report publisher
-func NewReportPublisher(publisher Publisher) *ReportPublisher {
+func NewReportPublisher(publisher Publisher, noControls bool) *ReportPublisher {
 	return &ReportPublisher{
-		publisher: publisher,
+		publisher:  publisher,
+		noControls: noControls,
 	}
 }
 
+func removeControls(r report.Report) report.Report {
```

Comment:
Yep, I thought about that too. It would be easy to forget to add new topologies here.

The exiting listing function copies the topologies, so it won't let us modify them in the report.

I will add a MapTopologies function.